test/AutoRest.TestServer.Tests/Infrastructure/TestServerBase.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace AutoRest.TestServer.Tests.Infrastructure
+{
+    public class TestServerBase : IDisposable
+    {
+        private static Lazy<BuildPropertiesAttribute> _buildProperties = new(() => (BuildPropertiesAttribute)typeof(TestServerBase).Assembly.GetCustomAttributes(typeof(BuildPropertiesAttribute), false)[0]);
+
+        private readonly Process _process;
+        public HttpClient Client { get; }
+        public Uri Host { get; }
+
+        public TestServerBase(string baseDirectory, string processArguments)
+        {
+            var portPhrase = "Started server on port ";
+            var startup = Path.Combine(baseDirectory, "dist", "cli", "cli.js");
+
+            var processStartInfo = new ProcessStartInfo("node", $"{startup} {processArguments}")
+            {
+                // Use random port
+                Environment = { ["PORT"] = "0" },
```

Comment:
Not sure why we set port=0 here but the comment mentioned 'use random port'

Reply:
Correct.